### ISSUE TITLE
How to get cardtype from jquery?

### ISSUE THREAD

**Issue by fabiowoj** [jessepollak/card#412]
How to get cardtype from jquery to fill an input form with that cardtype? 
I know many messages but not work from jquery. 
Thanks

**Comment by luizdesign** [comment]
Hi @fabiowoj, me too needed to do it. I resolved using  the Payment object, it's a Card submodule.
```js
const cards = Payment.getCardArray(); // Returns the complete cards validation
for (let i = 0; i < cards.length; i += 1) {
  if (cards[i].pattern.test(creditCardNumber)) {
    cards[i].type; // the credit card type
  }
}
```
Where creditCardNumber is a variable with the entered credit card number.

You can use also:
- Payment.getCardArray().filter(...)
- Select the card element in DOM and remove other classes (i do not recommended)

**Comment by fabiowoj** [comment]
Thanks @luizdesign 
I'll try it.

**Comment by QuintonC** [comment]
This issue should be marked as closed as @luizdesign provided a solution.

**Comment by morgan2233** [comment]
Sorry im a complete nub,

How would i use this code to fix my issue? would i put it inside the .js file

**Comment by chemmett** [comment]
You don't even have to implement this function, it's already in the Payment module.
`Payment.fns.cardType(creditCardNumber)`
It would be nice if an event was fired when the card type changed, but this definitely works.

**Comment by melloware** [comment]
This was published to NPM in 2.5.3
